### PR TITLE
feat: add `isSponsored` prop for `Swap`

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**
- add `isSponsored` prop

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: src/swap/components/SwapProvider.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ export function SwapProvider({
     maxSlippage: DEFAULT_MAX_SLIPPAGE,
   },
   experimental,
+  isSponsored,
```
Comment (cpcramer) [inline src/swap/components/SwapProvider.tsx:51]: default to `false`?

Comment (0xAlec) [reply src/swap/components/SwapProvider.tsx:51]: default for booleans are false

File: src/swap/components/Swap.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ export function Swap({
   onError,
   onStatus,
   onSuccess,
+  isSponsored,
```
Comment (alessey) [inline src/swap/components/Swap.tsx:24]: could default to false here and not have to handle undefined

Comment (0xAlec) [reply src/swap/components/Swap.tsx:24]: I don't think we need special handling for undefined since it's falsey

Comment (0xAlec) [reply src/swap/components/Swap.tsx:24]: but I can add a default here

File: src/swap/components/SwapProvider.tsx (hunk 1 above)
Comment (0xAlec) [reply src/swap/components/SwapProvider.tsx:51]: added in latest diff

File: src/swap/components/Swap.tsx (hunk 2 above)
Comment (0xAlec) [reply src/swap/components/Swap.tsx]: added in latest diff

Comment (cpcramer) [reply src/swap/components/Swap.tsx]: I still like explicitly defining as false

File: src/swap/types.ts [hunk 3]
```diff
@@ -267,6 +269,7 @@ export type SwapProviderReact = {
   experimental: {
     useAggregator: boolean; // Whether to use a DEX aggregator. (default: true)
   };
+  isSponsored?: boolean; // Whether the swap is sponsored. (default: false)
```
Comment (cpcramer) [inline src/swap/types.ts:272]: nit: starting the comment description with `// An optional...`

File: src/swap/components/Swap.tsx [hunk 4]
```diff
@@ -48,6 +49,7 @@ export function Swap({
       onError={onError}
       onStatus={onStatus}
       onSuccess={onSuccess}
+      isSponsored={isSponsored}
```
Comment (cpcramer) [inline src/swap/components/Swap.tsx:52]: nit: alphabetic imports

File: src/swap/components/Swap.tsx [hunk 5]
```diff
@@ -21,6 +21,7 @@ export function Swap({
   onError,
   onStatus,
   onSuccess,
+  isSponsored = false,
```
Comment (cpcramer) [inline src/swap/components/Swap.tsx:24]: nit: alphabetic

File: src/swap/types.ts (hunk 3 above)
Comment (0xAlec) [reply src/swap/types.ts]: done

File: src/swap/types.ts [hunk 6]
```diff
@@ -282,6 +285,7 @@ export type SwapReact = {
   experimental?: {
     useAggregator: boolean; // Whether to use a DEX aggregator. (default: true)
   };
+  isSponsored?: boolean; // Whether the swap is sponsored. (default: false)
```
[the remark this answers is not in the document]
Comment (0xAlec) [reply src/swap/types.ts]: done

File: src/swap/components/Swap.tsx (hunk 4 above)
Comment (0xAlec) [reply src/swap/components/Swap.tsx]: done - this was from when we called it `sponsored`

File: src/swap/components/Swap.tsx (hunk 5 above)
Comment (0xAlec) [reply src/swap/components/Swap.tsx]: done - this was from when we called it `sponsored`

**Review by cpcramer — APPROVED** [review]

**Pull request merged**